macOS/debugger.cpp

```diff
@@ -68,12 +68,78 @@ vm_prot_t Debugger::MacOSProtectionFlags(MemoryProtection memory_protection) {
   }
 }
 
+void Debugger::ClearSharedMemory() {
+  for (auto iter = shared_memory.begin(); iter != shared_memory.end(); ) {
+    iter = FreeSharedMemory(iter);
+  }
+
+  shared_memory.clear();
+}
+
+void Debugger::FreeSharedMemory(SharedMemory sm) {
```

Comment:
I think you misunderstood me here, what I meant is that, instead of duplicating code between this and `FreeSharedMemory(std::list<SharedMemory>::iterator it)`, `FreeSharedMemory(std::list<SharedMemory>::iterator it)` would call this one as a helper function and this function wouldn't touch the `shared_memory` list at all.

As it stands, this function is never called and can just be removed.